lib/src/klio/transforms/decorators.py

```diff
@@ -525,8 +518,17 @@ def method_wrapper(self, *args, **kwargs):
                         "incoming_item": incoming_item,
                         "metrics": metrics_objs,
                     }
-
-                return wrapper(*args, **wrapper_kwargs, **kwargs)
+                    result = wrapper(*args, **wrapper_kwargs, **kwargs)
+                    yield from result
```

Comment:
have you confirmed that this still works for when `@handle_klio` decorates functions and non-generator methods? I know that whenever you put a `yield` in a function, it makes it a generator - even if you never hit the `yield` branch. I would assume this is the same case for `yield from`. This is why the existing `yield`'ing and `yield from`'ing are in their own functions.

Reply:
Decorating functions is unaffected since the other wrapper handles that (also tested and it seems fine).  For methods though, what other methods besides `process` would we want to use this decorator on?

Reply:
Wouldn't decorated functions be affected though, since this is `_handle_klio`? and adding `yield from` to `_handle_klio` makes it a generator?

To answer your question, `handle_klio`/`_handle_klio` is meant to work with any method or function. It just makes the simplistic assumption that `DoFn.process` is the only method that's a generator. It's meant to allow folks to have classes - if they so please - that are not necessarily transforms.

Reply:
ah okay sorry I see now - I just finally properly tested it - and you're right re: if it's just function. my mistake - so sorry.

But yeah this won't work if a class method is not a generator. As I said, someone could have a non-transform class that has a method decorated with `handle_klio`. But also, I just remembered, folks can have composite transforms.

Some cases `yield from` does not work:

```py
class MyClass:
    @decorators.handle_klio
    def my_method(self, item):
        print(f"MyClass.my_method got item: {item}")
        return item

my_class = MyClass()

def my_map_func(item):
    item = my_class.my_method(item)
    return item

def run (input_pcol, config):
    return input_pcol | beam.Map(my_map_func)
```

```py
def my_map_func(item):
    print(f"my_my_func got item: {item}")
    return item

class MyComposite(beam.PTransform):
    @decorators.handle_klio
    def expand(self, pcoll):
        return (
            pcoll | beam.Map(my_map_func)
        )

def run (input_pcol, config):
    return input_pcol | MyComposite()
```